### PR TITLE
Throw error on JavaCall.init if not on root Task and JULIA_COPY_STACKS is not set

### PR DESCRIPTION
**Pull request by mkitti**
This pull request creates new assertions for `JavaCall.init` and `JavaCall.destroy` which will throw an error if JULIA_COPY_STACKS is not set and the current `Task` is not the root `Task`.

Without JULIA_COPY_STACKS=yes, `JavaCall.init` would result in a segmentation fault for me. With the added assertion, I just get an error message now from which the Julia REPL can recover.

The assertion could be expanded for use with the other exported calls such as `jcall`. Currently trying to execute those from the REPL just results in a wrong return value than a segmentation fault.

### CODE REVIEW

File: src/jvm.jl [hunk 1]
```diff
@@ -170,6 +170,7 @@ assertnotloaded() = isloaded() ? throw(JavaCallError("JVM already initialised"))
 # Pointer to pointer to pointer to pointer alert! Hurrah for unsafe load
 function init(opts)
     assertnotloaded()
+	assertroottask()
```
Comment (aviks) [inline src/jvm.jl]: indentation

File: src/jvm.jl [hunk 2]
```diff
@@ -246,6 +247,7 @@ function destroy()
     if (!isdefined(JavaCall, :penv) || penv == C_NULL)
         throw(JavaCallError("Called destroy without initialising Java VM"))
     end
+	assertroottask()
```
Comment (aviks) [inline src/jvm.jl]: indentation

File: src/task.jl [hunk 3]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (aviks) [inline src/task.jl]: maybe add it to the jvm.jl. It gets difficult to read IMO if code is spread out into very small files.

**Comment by mkitti** [comment]
indentation: I will check my Atom / Juno settings.  I suspect it introduced an actual tab.

I'm not sure that the assertion is only needed in jvm.jl. It may also be needed for jcall and other functions. For example, if the JVM is started from a programfile by a file like in the tests,  jcall would still silently fail at the REPL Julia prompt by returning an incorrect value. 

Do you have a recommendation for another file that included first?

File: src/jvm.jl (hunk 2 above)
Comment (mkitti) [reply src/jvm.jl]: Fixed in 0614efa . There was another [tab](https://github.com/JuliaInterop/JavaCall.jl/pull/105/commits/0614efa89d1bc67e46bd80d96f9070af6a42f87c#diff-72e942232acb05e3dc6f606c0a05a86fL71) in the file as well.

Comment (mkitti) [reply src/jvm.jl]: I fixed additional indentation issues in JavaCall.jl due to tab characters: de0bfe5

File: src/task.jl (hunk 3 above)
Comment (mkitti) [reply src/task.jl]: I consolidated the new functions code in jvm.jl. I added the `JULIA_COPY_STACKS` global to JavaCall.jl and returned the evaluation to `__init__` which was necessary anyways.

**Pull request merged**